third_party/3/aiofiles/threadpool/binary.pyi

```diff
@@ -1,5 +1,43 @@
+from _typeshed import AnyPath, ReadableBuffer, WriteableBuffer
+from io import FileIO
+from typing import Generic, Iterable, List, Optional, TypeVar, Union
+
 from ..base import AsyncBase
 
-class AsyncBufferedIOBase(AsyncBase[bytes]): ...
-class AsyncBufferedReader(AsyncBufferedIOBase): ...
-class AsyncFileIO(AsyncBase[bytes]): ...
+_FileName = TypeVar("_FileName", bound=Union[AnyPath, int])
+
+class _UnknownAsyncBinaryIO(AsyncBase[bytes], Generic[_FileName]):
```

Comment:
I'm not sure making this generic over _Filename is worth it; it's slightly more precise for the `.filename` property, but now forces everyone who uses one of these classes as a type annotation to add a type argument.

Reply:
> now forces everyone who uses one of these classes as a type annotation to add a type argument.

Is it true? I use pyright and it doesn't complain about not setting a generic type.

```python
async def(file: AsyncTextIOWrapper):
    name = file.name
    # name is auto-detected to be of type: str | bytes | _PathLike[str] | _PathLike[bytes] | int
```

Of course, if other type checkers complain, I can annotate `name` property as `-> Union[AnyPath, int]`.

Reply:
Mypy does this if you use the `--disallow-any-generics` option.

Reply:
Ok, removed that.